src/components/geonode.jsx

```diff
@@ -63,10 +64,23 @@ class GeoNodeViewer extends React.Component {
     this.updateMap(this.props);
     this.mode = this.props.mode || 'viewer';
     this.edit = (this.mode === 'composer');
+    if (this.props.layer) {
+      WMSService.getCapabilities(this.props.wmsServer, this._getLayerInfo.bind(this), function() {}, this.props.proxy);
+    }
   }
   componentWillReceiveProps(props) {
     this.updateMap(props);
   }
+  _getLayerInfo(layerInfo, getMapUrl) {
```

Comment:
should we move that into a service maybe? 
or into the SDK? and call the SDK to get the info for a specific layer?